### PR TITLE
feat(openstack): add Openstack provider

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

This PR implements OpenStack provider support in the Prowler SDK/CLI, enabling security assessments for OpenStack cloud environments.

OpenStack is a widely-used open-source cloud computing platform, and this integration allows Prowler to perform security audits on OpenStack deployments, complementing the existing support for other cloud providers

### Description

This PR adds comprehensive OpenStack provider integration to Prowler with the following components:

  **Core Provider Implementation:**
  - New `OpenStackProvider` class with Keystone Identity v3 authentication
  - Environment variable support for credentials:
    - Required: `OS_AUTH_URL`, `OS_USERNAME`, `OS_PASSWORD`, `OS_PROJECT_ID`, `OS_REGION_NAME`
    - Optional: `OS_IDENTITY_API_VERSION`, `OS_USER_DOMAIN_NAME`, `OS_PROJECT_DOMAIN_NAME`
  - Smart project ID handling supporting both UUID format and provider-specific formats (e.g., OVH numeric IDs)
  - Static `test_connection()` method for credential validation without full provider initialization
  - Mutelist support for excluding specific OpenStack resources from checks
  - Integration with Prowler's output formats (JSON, CSV, HTML, etc.)

  **Services Implemented:**
  1. **Compute (Nova Service)** - Manages virtual machine instances

  **Security Checks:**
  1. `compute_instance_security_groups_attached` - Ensures compute instances have security groups attached for network protection

  **Additional Changes:**
  - Updated `CheckMetadataModel` to support OpenStack-specific fields
  - Added OpenStack output rendering in HTML reports
  - Extended CLI parser with OpenStack-specific arguments
  - Added OpenStack compliance framework placeholder structure
  - Fixed compliance loading issue where non-JSON files were being processed

  **Testing Coverage:**
  - **57 comprehensive unit tests** including:
    - Provider initialization with various credential combinations
    - Environment variable handling and validation
    - Session configuration for UUID and non-UUID project IDs
    - Connection testing with error scenarios
    - Identity enrichment with graceful failure handling
    - Service initialization and resource listing
    - Check execution with PASS/FAIL scenarios
    - Edge cases (missing attributes, SDK exceptions, empty resources)
  - All tests include environment isolation to prevent interference from local OpenStack configurations

  **Dependencies:**
  - Added `openstacksdk` package to pyproject.toml for OpenStack API interactions

### Steps to review

**1. Run Tests:**
```bash
poetry run pytest tests/providers/openstack/ -v
# Expected: 57 tests passed
```

**2. Verify Provider Initialization:**
```bash
# Set OpenStack credentials (use your test environment)
export OS_AUTH_URL="https://your-openstack:5000/v3"
export OS_USERNAME="your-username"
export OS_PASSWORD="your-password"
export OS_PROJECT_ID="your-project-id"
export OS_REGION_NAME="RegionOne"

# List available services and checks
poetry run prowler openstack --list-services
poetry run prowler openstack --list-checks
```

**3. Test Credential Validation:**
```bash
# Should fail with clear error message about missing credentials
unset OS_AUTH_URL
poetry run prowler openstack

# Should succeed with valid credentials
export OS_AUTH_URL="https://your-openstack:5000/v3"
poetry run prowler openstack --check keystone_projects_enabled
```

**4. Execute Security Checks:**
```bash
# Run all OpenStack checks
poetry run prowler openstack

# Run by service
poetry run prowler openstack --service keystone
poetry run prowler openstack --service compute

# Run specific check
poetry run prowler openstack --check compute_instance_security_groups_attached
```

**5. Verify Code Quality:**
- Check that all new code follows Google Python Style Guide docstrings
- Verify type hints are present on all public functions
- Review error handling and logging patterns
- Confirm test coverage for edge cases

**6. Review Key Files:**
- `prowler/providers/openstack/openstack_provider.py` - Provider implementation
- `prowler/providers/openstack/models.py` - Session and identity models
- `prowler/providers/openstack/services/*/` - Service implementations
- `tests/providers/openstack/*` - Test coverage

### Checklist

<details>

<summary><b>Community Checklist</b></summary>

- [ ] This feature/issue is listed in [here](https://github.com/prowler-cloud/prowler/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or roadmap.prowler.com
- [ ] Is it assigned to me, if not, request it via the issue/feature in [here](https://github.com/prowler-cloud/prowler/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or [Prowler Community Slack](goto.prowler.com/slack)

</details>

- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.
- [ ] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/prowler/CHANGELOG.md), if applicable.

#### SDK/CLI
- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.

#### UI
- [ ] All issue/task requirements work as expected on the UI
- [ ] Screenshots/Video of the functionality flow (if applicable) - Mobile (X < 640px)
- [ ] Screenshots/Video of the functionality flow (if applicable) - Table (640px > X < 1024px)
- [ ] Screenshots/Video of the functionality flow (if applicable) - Desktop (X > 1024px)
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/ui/CHANGELOG.md), if applicable.

#### API
- [ ] All issue/task requirements work as expected on the API
- [ ] Endpoint response output (if applicable)
- [ ] EXPLAIN ANALYZE output for new/modified queries or indexes (if applicable)
- [ ] Performance test results (if applicable)
- [ ] Any other relevant evidence of the implementation (if applicable)
- [ ] Verify if API specs need to be regenerated.
- [ ] Check if version updates are required (e.g., specs, Poetry, etc.).
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/api/CHANGELOG.md), if applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by andoniaf** [comment]
## Local tests 
image

File: prowler/providers/openstack/services/compute/compute_instance_security_groups_attached/compute_instance_security_groups_attached.metadata.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "Provider": "openstack",
+  "CheckID": "compute_instance_security_groups_attached",
+  "CheckTitle": "Compute instances have security groups attached",
+  "CheckType": [
+    "Software and Configuration Checks/OpenStack Security Best Practices",
+    "Software and Configuration Checks/Industry and Regulatory Standards",
+    "TTPs/Initial Access"
+  ],
```
Comment (andoniaf) [inline prowler/providers/openstack/services/compute/compute_instance_security_groups_attached/compute_instance_security_groups_attached.metadata.json]:
```suggestion
  "CheckType": [],
```

Comment (andoniaf) [reply prowler/providers/openstack/services/compute/compute_instance_security_groups_attached/compute_instance_security_groups_attached.metadata.json]: [This field is only applicable to the AWS provider.](https://docs.prowler.com/developer-guide/checks#checktype)

File: prowler/providers/openstack/lib/mutelist/mutelist.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class OpenStackMutelist(Mutelist):
+    """Mutelist implementation for the OpenStack provider."""
+
+    def is_finding_muted(
+        self,
+        finding: CheckReportOpenStack,
+        project_id: str,
+    ) -> bool:
```
[the remark this answers is not in the document]
Comment (danibarranqueroo) [reply prowler/providers/openstack/lib/mutelist/mutelist.py:13]: Not needed

File: prowler/providers/openstack/openstack_provider.py [hunk 3]
```diff
@@ -0,0 +1,518 @@
+from os import environ
+from pathlib import Path
+from typing import Optional
+
+import openstack
```
[the remark this answers is not in the document]
Comment (danibarranqueroo) [reply prowler/providers/openstack/openstack_provider.py]: Done

File: prowler/providers/openstack/services/compute/compute_instance_security_groups_attached/compute_instance_security_groups_attached.metadata.json (hunk 1 above)
Comment (danibarranqueroo) [reply prowler/providers/openstack/services/compute/compute_instance_security_groups_attached/compute_instance_security_groups_attached.metadata.json]: Good catch, thanks!

**Review by andoniaf — APPROVED** [review]

**Review by cesararroba — APPROVED** [review]
🚀
